Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm

```diff
@@ -639,6 +639,28 @@ HTTPServer server({
     EXPECT_WK_STREQ([opened.uiDelegate waitForAlert], "false");
 }
 
+TEST(SiteIsolation, LoadDuringOpen)
+{
+    HTTPServer server({
+        { "/example"_s, { "window.open('https://webkit.org/webkit')"_s } },
+        { "/webkit"_s, { "hi"_s } },
+    }, HTTPServer::Protocol::HttpsProxy);
+    auto [webView, navigationDelegate] = siteIsolatedViewAndDelegate(server);
+    RetainPtr uiDelegate = adoptNS([TestUIDelegate new]);
+    uiDelegate.get().createWebViewWithConfiguration = [&](WKWebViewConfiguration *configuration, WKNavigationAction *action, WKWindowFeatures *windowFeatures) -> WKWebView * {
+        RetainPtr auxiliary = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectZero configuration:[webView configuration]]);
+        auxiliary.get().navigationDelegate = navigationDelegate.get();
+        [auxiliary loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"https://example.com/example"]]];
+        [navigationDelegate waitForDidFinishNavigation];
+        return nil;
+    };
```

Comment:
My understanding with this approach is:
- view1 loads example.com and requests a new view
- `createWebViewWithConfiguration` is invoked with configuration that has opener iInfo
- Client creates view2 with the configuration, which clears opener info in configuration and set the opener info on view2 (`WebPageProxy::m_browsingContextGroup`)
- view2 loads example.com and it will be loaded in a different process than view1
- `createWebViewWithConfiguration` returns

My question is what makes view2 load in a different process when it gets the opener info from the configuration? (Maybe I am missing something)

Reply:
If view 2 is created from the provided configuration in createWebViewWithConfiguration then it will share the same BrowsingContextGroup with view 1 and will load example.com in the same process as view 1 has loaded example.com in.  However, in this test auxiliary is created with a copy of view 1's configuration, not the provided configuration, so it needs to have a different BrowsingContextGroup.

Reply:
I see! Do we need to support the case where client creates view2 with the provided configuration, and loads example.com? (Will this work without site isolation?)
Based on your investigation, maybe we should set a flag on `WebProcessProxy` when it's blocked waiting on a sync message reply and check the flag when the `WebProcessProxy` is picked for loading.

Reply:
No, that never worked even without site isolation.